File: 242-valid-anagram/valid-anagram.kt
class Solution {
    fun isAnagram(s: String, t: String): Boolean {
        if(s.length != t.length) return false
        
        val map = mutableMapOf<Char,Int>()


        for(i in s){
            map[i] = map.getOrDefault(i,0) + 1
        }

         for(i in t){
            map[i] = map.getOrDefault(i,0) - 1
        }

        for(i in map.values) if (i != 0) return false
        return true


    }
}

File: 242-valid-anagram/valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
            return False

        seen = [0]*26
        for i,j in zip(s,t):
            seen[ord(i) - ord("a")] += 1
            seen[ord(j) - ord("a")] -= 1
        
        for i in seen:
            if i != 0:
                return False
        return True
        
            
        